org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenBuildSupport.java

```diff
@@ -26,23 +41,87 @@
  */
 public class MavenBuildSupport implements IBuildSupport {
 
+	private static final String POM_SERLIZATION_FILE_NAME = ".pom-digests";
```

Comment:
I really don't like the fact that we're using a static map here. I guess it comes from the fact that MavenBuildSupport is used as a stateless object. I'd rather change that and keep the map in the instance.
Also, we should use the workspace state location available to eclipse plugins instead of using the workspace root.